returnn/tf/layers/basic.py

```diff
@@ -6301,6 +6304,20 @@ def __init__(
         else:
             x = input_data.placeholder
 
+        if pad_seq_len_to_power is not None:
+            padding_for_power = []
+            for ax in range(input_data.batch_ndim):
+                if input_data.is_axis_dynamic(ax):
```

Comment:
In your previous test, I think you only padded the spatial dims, not the batch dim. Is this intended that you also pad the batch dim here?

Reply:
Oh you are right. `is_axis_dynamic` is a bit confusing.

Reply:
In your test, it was already constant mem consumption (after a while of running) when only doing this?

Reply:
I started a training for my real config with this option in the first two layers. However, there are more `ConvLayers`. I could add it to all or I could also create a config for testing e.g. with an adapted `Task12AXDataset`.

Reply:
Is it needed for all? This was not clear yet, or was it? Or actually I thought in your other network where you only have the other layers, this problem does not occur?

Reply:
I think the other layers also contribute to the memory increase. However, since they operate on subsampled sequences, the number of different input shape is not so large, so the increase is not as significant as on the waveform.

Reply:
Ah ok. So if this is a problem for all ConvLayers, instead of having this a per-layer option, this could also be a global option in the config.

Reply:
How would you do that? Can I access the global config from within a layer?

Reply:
`self.network._config` might work but is there a nice way to access it?

Reply:
Something like `pad_seq_len_to_power = pad_seq_len_to_power or self.network.get_config().float("conv_pad_seq_len_to_power", None)`?

Reply:
Yes `self.network.get_config()`. But I think I would also remove the layer option completely then (unless you think it makes sense to also do this per layer).

Reply:
It depends a bit on how much this affects the performance I guess. Also, you might want to specify lower values here for initial layers (where sequence lengths are tens of thousands) and higher values for layers after subsampling.

Reply:
It's logarithmic, so shouldn't it be the same?
But yes ok, if you think it's useful for you, then add it.

Reply:
It only adds minimally to the code complexity, so I think I'd keep it.